### ISSUE TITLE
Front End: Full VBA regional subsection content

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#14957]
## Description or Additional Context 
Some services accordions are at a regional level. See [Slack thread](https://dsva.slack.com/archives/C0FQSS30V/p1695912435197179)

### Resources
- VBA Content Specs listed in #13096 
- [Sketch file](https://www.sketch.com/s/891d33ae-152d-471c-ab5d-9aedf89cf6ff)

## Acceptance Criteria
- [x] Confirm all needed data from the query are brought in.
- [x] The front end shown matches the design file to a reasonably accurate extent.
- [x] Any applicable unit or component testing is completed.
- [x] Any refinements to the front end have been stubbed out in tickets.
- [x] Requires design review
- [x] Requires accessibility review

## Team
Please check the team(s) that will do this work.

- [ ] CMS Team
- [ ] Public Websites
- [X] Facilities
- [ ] User support
- [ ] Accelerated Publishing

**Comment by xiongjaneg** [comment]
@xiongjaneg create follow up ticket similar to Facility ticket

**Comment by swirtSJW** [comment]
To connect the facility query to get the service region, you need to include a reverse entity ref lookup on this field
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/dda73261-f4d2-4394-93da-2cb2bc041bde)

**Comment by swirtSJW** [comment]
Actually  I need to follow up with @davidmpickett because I think my pointer to the vba_region_facility_list might be incorrect.  I won't know for sure until I understand where this data lives on the comps.  
[Slack discussion](https://dsva.slack.com/archives/C0FQSS30V/p1695912435197179)

**Comment by xiongjaneg** [comment]
Screenshot from 16th minute
<img width="532" alt="Screenshot 2023-09-28 at 10 27 55 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/73648238/6b98e46d-4139-44ce-8dc7-4aded0d41040">

**Comment by swirtSJW** [comment]
Followup from 16th minute.  

- Service location from Service region display (see screenshot above) should come after service locations that come from services.
-Service location is going to come from the service.  There are two possible ways to get the service location.  I will outline each of them in a few minutes.

**Comment by xiongjaneg** [comment]
Blocked in sprint 94 per [Slack thread](https://dsva.slack.com/archives/C0FQSS30V/p1695925683286289?thread_ts=1695912435.197179&cid=C0FQSS30V)

**Comment by xiongjaneg** [comment]
Still blocked by VBA facility being started to add to

**Comment by jilladams** [comment]
From 1:1 today, @maxx1128  noted he's gotten Jordan's review via a screencap video ([Slack thread](https://dsva.slack.com/archives/C0FQSS30V/p1706206533288999)), plans to tag @laflannery for review in a Tugboat, and is feeling this may be ready to PR/demo next week.

@maxx1128 pls update if I misstated anything there.

**Comment by maxx1128** [comment]
@jilladams, this looks accurate to me 👍 My upcoming tugboat instance will also have the service region accordions updated to V3.

**Comment by maxx1128** [comment]
Update for end of Sprint 102: this ticket is blocked due to some expected merge conflicts with @eselkin's work on facility and regional services. Once that is set, the accessibility review is complete, and any other needed changes are made, this should be able to be completed in the next sprint.

**Comment by laflannery** [comment]
Headings:
- The first heading in each accordion right now is an h3 but should be an h4. I'm not sure if this is editor content that need to be fixed in a text editor or in the template: "Get help checking the status of your COE", "Vascular Surgery Regional Service Header", "Submit your application by mail", etc.
- The heading for "Can't find the service you're looking for?" there is an outstanding question on what level of heading this should be. @mmiddaugh brought this up in another ticket that I can't find and I would like to know her opinion on h2 or h3 here. The short version is that it should be an h2 in the hierarchy because "Other Services" might not always be present but should we be doing some sort of conditional so that it's at H3 in most cases UNLESS it shouldn't be and therefore won't show in the ToC for the majority of cases?

**Comment by laflannery** [comment]
@maxx1128 I have finished my accessibility review, I know you already started looking at/updating based on my comments. I added a note about the telephone numbers to my original comment. I also found the [original conversation](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16901#issuecomment-1900771547) about the "Can't find the service you are looking for" heading. We don't have an answer quite yet but I did sync with Michelle and this is on her radar. I would hold on this for now until we get direction from her.

**Comment by laflannery** [comment]
@maxx1128 Final decision on that second header comment is that we have decided to leave "Can't find the service you are looking for" as an h2 - it is a sibling of the other service headings and we should treat it as such. If that means is has to be in the "On this page" component, that's ok

**Comment by jilladams** [comment]
https://dsva.slack.com/archives/C0FQSS30V/p1706639135108009
> tests pass after making banner testing not dependent on the whole page rendering with old data that doesn't match with the way the new data renders.

A11y feedback implemented and available in Tugboat (fyi @laflannery ) for verification, which will happen over S103 boundary.

**Comment by laflannery** [comment]
@maxx1128 I just reviewed the tugboat again and the headings look good

Some comments on Phone numbers:
- Is the `aria-describedby` dynamic? Because I have now realized that the editor can set this as the label in the CMS so do we pull this from there so that when they set the label, it will pull as the `aria-describedby`?
- There are 2 other props that are available in the component `sms` and `tty` these should be used when appropriate as well. You have an example of each in your tugboat but are we able to have these set up so whenever the editor choose to add these it will be set correctly?
   - Veterans Pension SMS number
   - Veterans Pension TTY number
- I also noticed that it seemed like not all numbers were update with the `aria-describedby` attribute. Can you double check that this is set to appear for all numbers (main location, VA Hotline, all service accordions, etc.)

**Comment by maxx1128** [comment]
@laflannery Looking at these points again this morning:

1. `aria-describedby` is dynamic. If there's a custom label set in the CMS for a number, that's what's used in the screen reader text.
2. Whenever an editor adds an `sms` and `tty` number, they're used along with any other numbers with the same properties and styling. Is there anything specific or different that needs to be done with these types of numbers? In the CMS they have the same list of properties.
3. I double-checked and saw that I missed adding `aria-describedby` for when it's the facility's phone number. I will add that in now for the next tugboat release.

**Comment by maxx1128** [comment]
@laflannery That's correct, as things are right now, the SMS and TTY numbers render the same code as the other types. I'll look into getting these changes made for these specific types of numbers.

**Comment by laflannery** [comment]
@maxx1128 We might want to hold on that and ask @xiongjaneg because it might be out of scope and not MVP for this particular ticket

**Comment by maxx1128** [comment]
Changing the component based on these types of numbers is just an extra property. I'll try updating the tugboat, it may be an easy win.

<img width="1112" alt="Screen Shot 2024-01-31 at 1 19 46 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/10790736/5848b7b4-06ef-469b-b631-9b1a871c57b5">

**Comment by maxx1128** [comment]
@laflannery looks like it worked! The extra properties appear on `sms` and `tty` numbers but not the others, and they update the markup as expected for each.

<img width="1579" alt="Screen Shot 2024-01-31 at 3 11 12 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/10790736/4221e94d-9296-4533-96aa-8e22253dce7a">

**Comment by laflannery** [comment]
@maxx1128 Regional subsections look swell! :)

**Comment by jilladams** [comment]
Closing. This has been in prod for end of Sprint 103, verified in staging envs, and will require prod content for prod verification. We can cut new tickets for any danglers.
